runtime/interpreter/big.go

```diff
@@ -140,3 +141,31 @@ func BigEndianBytesToSignedBigInt(b []byte) *big.Int {
 func BigEndianBytesToUnsignedBigInt(b []byte) *big.Int {
 	return new(big.Int).SetBytes(b)
 }
+
+func BigIntSqrt(interpreter *Interpreter, value *big.Int, locationRange LocationRange) UFix64Value {
+	if value.Sign() < 0 {
+		panic(UnderflowError{
+			LocationRange: locationRange,
+		})
+	}
+
+	valueFloat := new(big.Float).SetPrec(256).SetInt(value)
```

Comment:
I am making an assumption here:
```
By protocol convention, Cadence integer values are guaranteed to
floating-point value with 256 bit precision _without_ truncation. 
```
If that holds, I would suggest to explicitly document it:
```suggestion
	// By protocol convention, Cadence integer values are guaranteed to fit into a
	// floating-point value with 256 bit precision _without_ truncation. 
	valueFloat := new(big.Float).SetPrec(256).SetInt(value)
```

If that doesn't hold, we need to look at the rounding convention (e.g. adopting the [conventions from C](https://wiki.sei.cmu.edu/confluence/display/c/FLP34-C.+Ensure+that+floating-point+conversions+are+within+range+of+the+new+type))

Reply:
This is a very nice call-out. Made me realize a missing check. Thanks!

> By protocol convention, Cadence integer values are guaranteed to fit
into floating-point values with 256 bit precision _without_ truncation. 

The assumption is true for most Cadence integer types but not for `Int` and `UInt` which are arbitrary precision numbers.

That being said, since the output of `Sqrt` is a UFix64 (max value`184467440737.09551615`), the maximum argument value that the `Sqrt` function can work on is `34028236692093846346337`. See this calculation in `runtime/math_test.go`.

So, we need two things:
1. `Sqrt` function should validate that the input integer is <=  `34028236692093846346337`. I haven't done that yet.
2. Once the limit is in place, we can safely state that all Cadence integer values **supported by the Sqrt function** are guaranteed to fix into floating-point values with 256 bit precision without truncation.

cc: @turbolent FYI

Reply:
Added the overflow check and the comment in b2f799b13.